src/opa.js

```diff
@@ -49,13 +48,58 @@ function _dumpJSON(wasmInstance, memory, addr) {
     return JSON.parse(s);
 }
 
-
-function builtinPlus(a, b) {
-    return a+b;
-}
-
 const builtinFuncs = {
-    plus: builtinPlus,
+    // Numbers
+    plus: (a, b) =>  a + b,
+    minus: (a, b) => a - b,
+    mul: (a, b) => a * b,
+    div: (a, b) => a / b,
+    rem: (a, b) => a % b,
+    round: (x) => Math.round(x),
+    abs: (x) => Math.abs(x),
+
+    // Aggregates
+    count: (arr) => arr.length,
+    sum: (arr) => arr.reduce((a,b) => a + b, 0),
+    product: (arr) => arr.reduce((total, num) => total * num, 1),
+    max: (arr) => Math.max(arr),
+    min: (arr) => Math.min(arr),
+    sort: (arr) => [...arr].sort(),
+    all: (arr) => arr.length === 0 ? true : arr.every(v => v === true),
+    any: (arr) => arr.length === 0 ? false : arr.includes(true),
+    
+    // Arrays
+    "array.concat": (arr1 , arr2) => arr1.concat(arr2),
+    "array.slice": (arr, startIndex, stopIndex) => arr.slice(startIndex, stopIndex),
+
+    // Strings
+    contains: (string, search) => string.includes(search),
```

Comment:
Thinking through some of this I wonder how strict we need to be about type checking. In the builtin implementations for OPA in Golang they pretty much all use a set of shared helpers to get operands and verify the type is what was expected (eg, passing a number or array into this one as `string` would be caught as invalid and an error raised).

My main concern is that we should have consistent behavior for builtin functions between normal OPA, Golang WASM runtime w/ OPA binary, and JS WASM runtime w/ OPA binary, etc. The same policy+data+query+input should give the same result on all of them.